From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: IllegalArgument <awilley1@gmail.com>
Date: Mon, 24 Apr 2023 12:56:13 -0400
Subject: [PATCH] 5261: Eagerly destroy stack placeholder definition op

If the stack placeholder varnode is unused after being removed as an input to its corresponding CALL op, delete its definition op as well. This prevents the placeholder varnode from confusing later phases of analysis with a spurious reference to the stack.
---
 Ghidra/Features/Decompiler/src/decompile/cpp/fspec.cc | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/Ghidra/Features/Decompiler/src/decompile/cpp/fspec.cc b/Ghidra/Features/Decompiler/src/decompile/cpp/fspec.cc
index c8044c1f4..2e50cbf75 100644
--- a/Ghidra/Features/Decompiler/src/decompile/cpp/fspec.cc
+++ b/Ghidra/Features/Decompiler/src/decompile/cpp/fspec.cc
@@ -4696,6 +4696,9 @@ void FuncCallSpecs::resolveSpacebaseRelative(Funcdata &data,Varnode *phvn)
     if (op->getIn(stackPlaceholderSlot) == phvn) {
       data.opRemoveInput(op,stackPlaceholderSlot);
       clearStackPlaceholderSlot();
+      if (phvn->hasNoDescend()) {
+        data.opDestroy(phvn->getDef());
+      }
       return;
     }
   }
-- 
2.40.0

